### PR TITLE
Add CapabilityRegistry skeleton

### PR DESCRIPTION
**Pull request by DeividasK**
TODOs:

- [ ] Split Forwarder changes (warning fixes) into a separate PR.

### CODE REVIEW

File: contracts/src/v0.8/keystone/CapabilityRegistry.sol [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        bytes32 capabilityId = getCapabilityID(capability);
+        capabilities[capabilityId] = capability;
+        emit CapabilityAdded(capabilityId);
+    }
+
+    function getCapability(bytes32 capabilityID) public view returns (Capability memory) {
+        return capabilities[capabilityID];
+    }
+
+    function getCapabilityID(Capability calldata capability) public pure returns (bytes32) {
```
Comment (DeividasK) [inline contracts/src/v0.8/keystone/CapabilityRegistry.sol:36]: Add comment as to why (fit into bytes32)

File: contracts/src/v0.8/keystone/CapabilityRegistry.sol [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
+    mapping(bytes32 => Capability) public capabilities;
+
+    event CapabilityAdded(bytes32 indexed capabilityId);
+
+    function typeAndVersion() external pure override returns (string memory) {
+        return "CapabilityRegistry 1.0.0";
+    }
+
+    function addCapability(Capability calldata capability) external onlyOwner {
```
Comment (DeividasK) [inline contracts/src/v0.8/keystone/CapabilityRegistry.sol:26]: Add validation for capability id to be unique.

File: contracts/src/v0.8/keystone/CapabilityRegistry.sol [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    bytes32 capabilityType;
+    // Semver, e.g., "1.2.3"
+    // bytes32(string); must be valid Semver + max 32 characters.
+    bytes32 version;
+}
+
+contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
+    mapping(bytes32 => Capability) public capabilities;
+
+    event CapabilityAdded(bytes32 indexed capabilityId);
```
Comment (cds95) [inline contracts/src/v0.8/keystone/CapabilityRegistry.sol:20]: Do we want to add natspec comments now?

File: contracts/src/v0.8/keystone/CapabilityRegistry.sol [hunk 4]
```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {TypeAndVersionInterface} from "../interfaces/TypeAndVersionInterface.sol";
+import {OwnerIsCreator} from "../shared/access/OwnerIsCreator.sol";
+
+struct Capability {
```
Comment (cds95) [inline contracts/src/v0.8/keystone/CapabilityRegistry.sol:7]: Should this be in a separate `ICapabilityRegistry` interface?

File: contracts/src/v0.8/keystone/CapabilityRegistry.sol [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    // bytes32(string); validation regex: ^[a-z0-9_\-:]{1,32}$
+    // Not "type" because that's a reserved keyword in Solidity.
+    bytes32 capabilityType;
+    // Semver, e.g., "1.2.3"
+    // bytes32(string); must be valid Semver + max 32 characters.
+    bytes32 version;
+}
+
+contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
+    mapping(bytes32 => Capability) public capabilities;
```
Comment (cds95) [inline contracts/src/v0.8/keystone/CapabilityRegistry.sol:18]:
```suggestion
    mapping(bytes32 => Capability) private s_capabilities;
```

To match Solidity style guide

**Pull request closed without merging**